File: leetcode/C++/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l=0, r=(int)nums.size()-1,m=0 , sol=0;
        while(l<=r){
            m=(l+r)/2;
            if(nums[m]<target){sol=l=m+1;}
            else if(nums[m]>target){sol=m; r=m-1;}
            else {sol=m; break;}
        }
        return sol;
    }
};

File: leetcode/Python/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r, m, sol = 0, len(nums)-1, 0, 0
        while l<=r:
            m = (l+r)//2
            if nums[m]<target:
                l, sol = m+1, m+1
            elif nums[m]>target:
                r, sol = m-1, m
            else:
                sol = m
                break
        return sol